dataprofiler/tests/profilers/test_base_option.py

```diff
@@ -6,56 +6,56 @@
-		with self.assertRaisesRegex(ValueError, expected_error):
-			options._set_helper("notadictionary", "")
-		with self.assertRaisesRegex(ValueError, expected_error):
-			options._set_helper(["not", "a", "dictionary"], "")
-
-		# Variable Path Is Not A String
-		expected_error = "The variable path must be a string."
-		with self.assertRaisesRegex(ValueError, expected_error):
-			options._set_helper({"hello": "world"}, 1)
-		with self.assertRaisesRegex(ValueError, expected_error):
-			options._set_helper({}, 1)
-
-	def test_set(self, *mocks):
-		options = BaseOption()
-
-		# Options Is Not A Dictionary
-		expected_error = "The options must be a dictionary."
-		with self.assertRaisesRegex(ValueError, expected_error):
-			options.set("notadictionary")
-		with self.assertRaisesRegex(ValueError, expected_error):
-			options.set(["not", "a", "dictionary"])
-	
-	def test_validate_helper(self, *mocks):
-		options = BaseOption()
-
-		with self.assertRaises(NotImplementedError):
-			options._validate_helper()
-	
-	def test_validate(self, *mocks):
-		options = BaseOption()
-
-		with self.assertRaises(NotImplementedError):
-			options.validate()
+class TestBaseOption(AbstractTestOptions, unittest.TestCase):
+    
+    @classmethod
+    def setUpClass(cls):
+        cls.option_class = BaseOption
+
+    def test_init(self, *mocks):
+        options = self.get_options()
+        self.assertDictEqual({}, options.properties) 
+
+    def test_set_helper(self, *mocks):
+        options = self.get_options()
+
+        # Options Is Not A Dictionary
+        expected_error = "The options must be a dictionary."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            options._set_helper("notadictionary", "")
+        with self.assertRaisesRegex(ValueError, expected_error):
+            options._set_helper(["not", "a", "dictionary"], "")
+
+        # Variable Path Is Not A String
+        expected_error = "The variable path must be a string."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            options._set_helper({"hello": "world"}, 1)
+        with self.assertRaisesRegex(ValueError, expected_error):
+            options._set_helper({}, 1)
+
+    def test_set(self, *mocks):
+        options = self.get_options()
+
+        # Options Is Not A Dictionary
+        expected_error = "The options must be a dictionary."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            options.set("notadictionary")
+        with self.assertRaisesRegex(ValueError, expected_error):
+            options.set(["not", "a", "dictionary"])
+    
+    def test_validate_helper(self, *mocks):
+        options = self.get_options()
+
+        with self.assertRaises(NotImplementedError):
+            options._validate_helper()
+    
+    def test_validate(self, *mocks):
+        options = self.get_options()
+
+        with self.assertRaises(NotImplementedError):
+            options.validate()
```

Comment:
If these are generic, I would recommend moving them to the abstract base class.

We are treating it more like a mixing for testing TBH.

Reply:
if not generic then fine to keep in place.

Reply:
They aren't generic, but every test class does need to implement these methods. I will at least have the methods raise a NotImplementedError if they aren't implemented by a test class.

Reply:
added in new commit